src/v2/Components/NavBar/NavItem.tsx

```diff
@@ -67,81 +127,92 @@ export const NavItem: React.FC<NavItemProps> = ({
     }
   }
 
+  const containerRef = useRef<null | HTMLDivElement>(null)
+
+  // Close the subnav if it is open and our focus moves outside of it
+  useEffect(() => {
+    const checkFocus = () => {
+      if (!containerRef?.current) return // Not mounted
+      if (!isVisible) return // Isn't open
+      if (containerRef.current.contains(document.activeElement)) return // Focus is within sub-nav
+
+      setIsVisible(false)
+    }
+
+    document.addEventListener("focus", checkFocus, true)
+    return () => {
+      document.removeEventListener("focus", checkFocus)
+    }
+  }, [isVisible])
+
+  const handleClick = () => {
+    onClick && onClick()
+    !!Menu && setIsVisible(prevIsVisible => !prevIsVisible)
+    trackClick()
+  }
+
+  // Add a very short delay to avoid trapping the mouse and capturing intent
+  const openDelayRef = useRef<null | ReturnType<typeof setTimeout>>(null)
+  // Add a slightly longer delay to improve UX when moving at an angle off HitArea
+  const closeDelayRef = useRef<null | ReturnType<typeof setTimeout>>(null)
+
+  const handleMouseEnter = () => {
+    openDelayRef.current = setTimeout(() => {
+      setIsVisible(true)
+    }, 50)
+    closeDelayRef?.current && clearTimeout(closeDelayRef.current)
+  }
+
+  const handleMouseLeave = () => {
+    openDelayRef?.current && clearTimeout(openDelayRef.current)
+    closeDelayRef.current = setTimeout(() => {
+      setIsVisible(false)
+    }, 150)
+  }
```

Comment:
This is intended as the replacement for the below: "Very hacky fix to prevent mouse out from triggering a close ..." Instead of positioning a div, just adding a simple delay so that when you mouse at a diagonal you can exit the hitarea for a brief moment then pick it back up. [There's nicer ways to do this](https://bjk5.com/post/44698559168/breaking-down-amazons-mega-dropdown) but this works a bit more consistently than the previous solution.

Reply:
Yah, this hack was a couple hours before launch kinda fix :/ Not ideal but it got the job done! 

This probably goes without saying, but just want to confirm: does this feel the same / better as the hack fix below? There are a lot of high value clicks at the far left of the full menu so want to be sure there's no UX degradation there.

Reply:
I think it feels better/about the same but I also had issues the previous solution and trying to get to the search input before.

![](http://static.damonzucconi.com/_capture/hKZKI20RkHay.gif)

I do think implementing a proper menu-aim-like thing here would be something we should do though.

Reply:
That screencap looks great 👍